pire/extra/count.cpp

```diff
@@ -22,13 +22,678 @@
 
 
 #include "count.h"
-#include "fsm.h"
+#include "../fsm.h"
 #include "../determine.h"
 #include "../glue.h"
 #include "../stub/lexical_cast.h"
 
 namespace Pire {
-	
+
+namespace Impl {
+
+typedef LoadedScanner::Action Action;
+typedef ymap<Char, Action> TransitionTagRow;
+typedef yvector<TransitionTagRow> TransitionTagTable;
+
+class CountingFsmTask;
+
+class CountingFsm {
+public:
+	typedef Fsm::LettersTbl LettersTbl;
+
+	enum {
+		NotMatched = 1 << 0,
+		Matched = 1 << 1,
+		Separated = 1 << 2,
+	};
+
+	CountingFsm(const Fsm& re, const Fsm& sep)
```

Comment:
Please consider marking constructors as explicit and accepting by value all arguments that you need copies of.

Reply:
You are right, I have done this.